game/common/game_board.py

```diff
@@ -0,0 +1,230 @@
+                then station_2 will be at (0, 0).
+
+        Lastly, another example will be shown to explain that you can combine both static and
+        dynamic assignments in the same dictionary:
+
+            locations =
+                {
+                    [vector_0_0] : [station_0],
+                    [vector_0_1] : [station_1],
+                    [vector_1_1, vector_1_2, vector_1_3] : [station_2, station_3, station_4]
+                }
+
+        In this example, station_0 will be at vector_0_0 without interference. The same applies to
+        station_1 and vector_0_1. However, for vector_1_1, vector_1_2, and vector_1_3, they will randomly
+        be assigned station_2, station_3, and station_4.
+
+    -----------------------------------------------------------------------------------------------------------
+
+    walled:
+        This is simply a bool value that will create a wall barrier on the boundary of the game_board. If
+        walled is True, the wall will be created for you.
+
+        For example, let the dimensions of the map be (5, 7). There will be wall Objects horizontally across
+        x = 0 and x = 4. There will also be wall Objects vertically at y = 0 and y = 6.
+
+        Below is a visual example of this, with 'x' being where the wall Objects are.
+
+        x x x x x   y = 0
+        x       x
+        x       x
+        x       x
+        x       x
+        x       x
+        x x x x x   y = 6
+    """
+
+    def __init__(self, seed: int | None = None, map_size: Vector = Vector(),
+                 locations: dict[[Vector]:[GameObject]] | None = None, walled: bool = False):
+
+        super().__init__()
+        self.seed = seed
+        random.seed(seed)
+        self.object_type: ObjectType = ObjectType.GAMEBOARD
+        self.event_active = None
+        self.map_size: Vector = map_size
+        self.locations: dict = locations
+        self.walled: bool = walled
+
+        # game_map is initially going to be None. Since generation is slow, call generate_map() as needed
+        self.game_map: list[list[GameObject]] | None = None
+
+    @property
+    def seed(self) -> int:
+        return self.__seed
+
+    @seed.setter
+    def seed(self, seed: int | None):
+        if seed is not None or not isinstance(seed, int):
+            raise ValueError("Seed must be an integer.")
+        self.__seed = seed
+
+    @property
+    def map_size(self) -> Vector:
+        return self.__map_size
+
+    @map_size.setter
+    def map_size(self, map_size: Vector):
+        if map_size is None or not isinstance(map_size, Vector):
+            raise ValueError("Map_size must be a Vector.")
+        self.__map_size = map_size
+
+    @property
+    def locations(self) -> dict:
+        return self.__locations
+
+    @locations.setter
+    def locations(self, locations: dict[[Vector]:[GameObject]] | None):
+        if locations is not None or not isinstance(locations, dict):
```

Comment:
Check if the len of each key is equal to the len of each value when setting as well